packages/ra-core/src/dataProvider/useDataProvider.ts

```diff
@@ -157,6 +158,13 @@ const useDataProvider = (): DataProviderProxy => {
                             'You must pass an onSuccess callback calling notify() to use the undoable mode'
                         );
                     }
+                    if (typeof enabled !== 'boolean') {
+                        throw new Error('The enabled option must be a boolean');
+                    }
+
+                    if (enabled === false) {
+                        return Promise.resolve();
```

Comment:
what if you resolve with an empty object `{}`?

Reply:
Yes this would solve the issue shown in the unit test. If this is fine enough for you, I can do this.

Reply:
That's what we do e.g. in [performOptimisticQuery](https://github.com/marmelab/react-admin/blob/a478d36fbadb507b5623fce3b1b25eca577f6baa/packages/ra-core/src/dataProvider/performQuery/performOptimisticQuery.ts#L118), so you're good to go.

Reply:
Nice then! I did the change and rebased my branch. 🙂